src/Features/Core/Portable/Workspace/BackgroundCompiler.cs

```diff
@@ -154,23 +149,13 @@ private void CancelBuild(bool releasePreviousCompilations)
 
             var logger = Logger.LogBlock(FunctionId.BackgroundCompiler_BuildCompilationsAsync, cancellationToken);
 
-            // Skip performing any background compilation for projects where user has explicitly
-            // set the background analysis scope to only analyze active files.
             var compilationTasks = allProjectIds
                 .Select(solution.GetProject)
                 .Select(async p =>
                 {
                     if (p is null)
                         return null;
 
-                    if (SolutionCrawlerOptions.GetBackgroundAnalysisScope(p) == BackgroundAnalysisScope.ActiveFile
-                        && p.Id != activeProject)
-                    {
-                        // For open files with Active File analysis scope, only build the compilation if the project is
-                        // active.
-                        return null;
-                    }
```

Comment:
❔ I'm assuming this was removed because the compiler analyzer is going to need this even if no other analyzers do?

Reply:
💭 Given I was not able to find any functional regression by removing BackgroundCompiler, I would also be fine with keeping this logic...

Reply:
Yes, same reasoning as in the prior comment. Compiler analyzer is going to walk through open files + the option should only be used for analyzer execution scope.